lib/Service/Classification/FeatureExtraction/IExtractor.php

```diff
@@ -45,11 +46,11 @@ public function prepare(Account $account,
 							array $messages): void;
 
 	/**
-	 * Return the feature value for the given sender address
+	 * Return the feature value for the given message
 	 *
-	 * @param string $email
+	 * @param Message $message
 	 *
-	 * @return float
+	 * @return float[]
```

Comment:
to take it another step further you could lift this to any kind of primitive rubix allows as input

Reply:
I like the idea. However, the Rubix DataSet class supports basically anything. We could refactor our code to use the DataSet class more generally, though.